src/tools/external.jam

```diff
@@ -0,0 +1,509 @@
+
+Allowed options:
+
+* `<include>`: the directory containing project's headers.
+* `<search>`: the directory containing project's library files.
+* `<name>`: should be in the form `LIBRARY=NAME`; overrides the searched name
+    for library `LIBRARY` with the name `NAME`. Multiple instances (even for
+    the same library) are allowed.
+
+The rule supports keyword arguments.
+
+Example:
+
+[source, jam]
+----
+external.configure sqlite3
+    : <include>/opt/sqlite3/include <search>/opt/sqlite3/lib
+    : <link>static
+    ;
+----
+
+|# # end::doc[]
+rule configure ( proj-name : options * : condition * )
+{
+    param.handle-named-params options condition ;
+
+    local ps = [ property-set.create $(condition) ] ;
+    local base = [ property-set.create [ $(ps).base ] ] ;
+    if $(ps) != $(base)
+    {
+        errors.user-error "Condition for external project '$(proj-name)'"
+            "contains free or incidental features:" [ $(ps).free ]
+            [ $(ps).incidental ] ;
+    }
+
+    local configs = [ get-configs $(proj-name) ] ;
+    register-config $(proj-name) : $(configs) : $(options) : $(ps) ;
+}
+
+#| tag::doc[]
+
+[[bbv2.tools.external.lib]]
+== Rule `lib`
+
+[source, jam]
+----
+rule lib ( target : sources * : options * )
+----
+
+The rule declares a target for external searched library. `target` argument
+should take the form `/PROJECT//LIBRARY`, which results in creation of target
+`LIBRARY` in project `PROJECT`. The `sources` argument can be used to
+represent transitive dependencies.
+
+Allowed options:
+
+* `<header>`: the header file (relative to include directory) that will be
+    used for include directory search. This argument is required.
+* `<name>`: provides alternative names for the searched library; multiple
+    instances are allowed. If omitted, the target name will be used.
+
+The rule supports keyword arguments.
+
+Example:
+
+[source, jam]
+----
+external.lib /sqlite3//sqlite3 : options <header>sqlite3.h ;
+exe myapp : /sqlite3//sqlite3 main.cpp ;
+----
+
+|# # end::doc[]
+rule lib ( tgt-id : sources * : options * )
+{
+    param.handle-named-params sources options ;
+    declare-target $(tgt-id) : $(sources) : $(options) ;
+}
+
+#| tag::doc[]
+
+[[bbv2.tools.external.alias]]
+== Rule `alias`
+
+[source, jam]
+----
+rule alias ( target : sources * : options * )
+----
+
+The rule declares a target for external header-only library. `target` argument
+should take the form `/PROJECT//LIBRARY`, which results in creation of target
+`LIBRARY` in project `PROJECT`. The `sources` argument can be used to
+represent transitive dependencies.
+
+Allowed options:
+
+* `<header>`: the header file (relative to include directory) that will be
+    used for include directory search. This argument is required.
+
+The rule supports keyword arguments.
+
+Example:
+
+[source, jam]
+----
+external.alias /catch2//libs : options <header>catch2/catch_version.hpp ;
+exe myapp : /sqlite3//sqlite3 main.cpp ;
```

Comment:
`/sqlite3//sqlite3` should be `/catch2//libs` here.

Reply:
Oops